modules/tests/shared/src/test/scala-3/io/circe/ConfiguredDerivesSuite.scala

```diff
@@ -504,3 +504,21 @@ class ConfiguredDerivesSuite extends CirceMunitSuite:
       assert(result === Right(tree), result)
     }
   }
+
+  {
+    given Configuration = Configuration.default.withSnakeCaseMemberNames
+
+    case class Bar(barValue: Int)
+    case class Foo(fooValue: Option[Bar]) derives ConfiguredEncoder
+
```

Comment:
Can you add a codec law here for `Foo` ?
```suggestion
checkAll("Codec[Foo]", CodecTests[Foo].codec)
```

Reply:
So I rolled out my solution approach onto the other type classes so that I can check codec laws (see new commit), but it turns out that "codec.encoder serializability" and "codec.decoder serializability" fail.

I will look into that when I find time, but if anyone has experience with that, any pointers would be welcome. (Been years since I last had to mess with java object serialization...)

Reply:
> So I rolled out my solution approach onto the other type classes so that I can check codec laws (see new commit), but it turns out that "codec.encoder serializability" and "codec.decoder serializability" fail.
> 
> I will look into that when I find time, but if anyone has experience with that, any pointers would be welcome. (Been years since I last had to mess with java object serialization...)

Nevermind, I fixed it.